modules/public/views.py

```diff
@@ -284,6 +320,35 @@ def external(attrs, new=False):
 
         formatted_paragraphs.append(paragraph.replace('\n', '<br />'))
 
+    country_data = []
+    try:
+        metrics_query_json = [
+            {
+                "metric_name": "installed_base_by_country_percent",
+                "snap_id": details['snap_id'],
+                "start": week_ago.strftime('%Y-%m-%d'),
+                "end": today.strftime('%Y-%m-%d')
+            }
+        ]
+
+        metrics_response = api.get_public_metrics(
+            snap_name,
+            metrics_query_json
+        )
+
+        users_by_country = normalize_metrics(
+            metrics_response[0]['series']
+        )
+        country_data = build_country_info(users_by_country)
+    except api.InvalidResponseContent as invalid_response_content:
+        message = str(invalid_response_content)
+        errors = [{'code': 'json-error', 'message': message}]
```

Comment:
Why do we need this structure to the error (`code` and `message`)? Why isn't it simply a string?

In any case, the `api` layer deliberately obfuscates the fact that it's using JSON, which is why it converts a `JSONDecodeError` into a `InvalidResponseContent` error (because we shouldn't care outside the API whether it's JSON or not). Why then are we inferring again that it is a JSON problem? If we care that it's a JSON error we should bubble that up from within the API layer. But my feeling is that we shouldn't care, in which case we shouldn't mention it here.

Reply:
@nottrobin we use this structure because:
- the server sends us this structure
- There could be mutlitple errors, so the frontend will iterate on those errors
- We were think about adding a mapping of all the errors to add our own messages since we have the list of possible [errors](https://dashboard.snapcraft.io/docs/api/snap.html#errors)